### PR TITLE
Handle aliases in comma-separated args

### PR DESCRIPTION
**Pull request by 0xmichalis**
@smarterclayton PTAL

### CODE REVIEW

**Comment by kargakis** [comment]
Should aliases be used in type/name syntax?

**Comment by smarterclayton** [comment]
Not for now.

On Jul 20, 2015, at 7:47 AM, Michail Kargakis <notifications@github.com>
wrote:

Should aliases be used in type/name syntax?

—
Reply to this email directly or view it on GitHub
<https://github.com/GoogleCloudPlatform/kubernetes/pull/11571#issuecomment-122859580>
.

File: pkg/kubectl/resource/builder.go [hunk 1]
```diff
@@ -298,11 +298,15 @@ func (b *Builder) ResourceTypeOrNameArgs(allowEmptySelector bool, args ...string
 
 func (b *Builder) replaceAliases(args []string) []string {
 	replaced := []string{}
-	for _, arg := range args {
-		if aliases, ok := b.mapper.AliasesForResource(arg); ok {
-			arg = strings.Join(aliases, ",")
+	for _, a := range args {
+		joinBack := []string{}
```
Comment (smarterclayton) [inline pkg/kubectl/resource/builder.go]: Nit: call this expanded

**Comment by kargakis** [comment]
@smarterclayton I have refactored `replaceAliases` a bit since it was parsing resource names and not only types so if someone had a resource named after an alias we would mess up
```
[vagrant@openshiftdev sample-app]$ oc get route all
Error from server: route "buildConfigs,builds,imageStreams,deploymentConfigs,replicationControllers,routes,services,pods" not found
```

There was no point in accepting more than one string since we always pass one argument as a type in the resource builder (expansion aside).

File: pkg/kubectl/resource/builder.go [hunk 2]
```diff
@@ -325,20 +329,6 @@ func hasCombinedTypeArgs(args []string) (bool, error) {
 	}
 }
 
-// ResourceTypeAndNameArgs expects two arguments, a resource type, and a resource name. The resource
-// matching that type and and name will be retrieved from the server.
-func (b *Builder) ResourceTypeAndNameArgs(args ...string) *Builder {
```
Comment (kargakis) [inline pkg/kubectl/resource/builder.go]: This is superceded by `ResourceTypeOrNameArgs`

**Comment by bgrant0607** [comment]
cc @jlowdermilk

File: pkg/kubectl/resource/builder.go [hunk 3]
```diff
@@ -296,16 +299,17 @@ func (b *Builder) ResourceTypeOrNameArgs(allowEmptySelector bool, args ...string
 	return b
 }
 
-func (b *Builder) replaceAliases(args []string) []string {
+// replaceAliases accept an argument and tries to expand any existing
```
Comment (smarterclayton) [inline pkg/kubectl/resource/builder.go]: nit: accepts an argument

File: pkg/kubectl/resource/builder.go (hunk 2 above)
Comment (smarterclayton) [inline pkg/kubectl/resource/builder.go]: They have different error conditions - ResourceTypeOrNameArgs does not require two arguments.  This method does.   Was ResourceTypeAndNameArgs not used anywhere now?

Comment (kargakis) [inline pkg/kubectl/resource/builder.go]: `ResourceTypeOrNameArgs` is literately a superset of this. Which is probably why this is not used anywhere now.

File: pkg/kubectl/resource/builder.go (hunk 1 above)
Comment (kargakis) [inline pkg/kubectl/resource/builder.go]: updated

File: pkg/kubectl/resource/builder.go (hunk 2 above)
Comment (smarterclayton) [inline pkg/kubectl/resource/builder.go]: Well, this returned an error if you didn't provide two arguments.  It's fine to remove.

**Comment by smarterclayton** [comment]
This is LGTM

File: pkg/kubectl/resource/builder_test.go [hunk 4]
```diff
@@ -942,13 +947,8 @@ func TestReplaceAliases(t *testing.T) {
 
 	for _, test := range tests {
 		replaced := b.replaceAliases(test.args)
```
Comment (jlowdermilk) [inline pkg/kubectl/resource/builder_test.go]: Shippable caught this, but you renamed the field to `arg`

Comment (kargakis) [inline pkg/kubectl/resource/builder_test.go]: fixed

**Comment by jlowdermilk** [comment]
lgtm

**Comment by mikedanese** [comment]
@k8s-bot ok to test

**Comment by mikedanese** [comment]
@k8s-bot ok to test

**Comment by mikedanese** [comment]
@k8s-bot ok to test

**Pull request merged**